### PR TITLE
Fixes to master branch and RFC 7517 compliance

### PR DESCRIPTION
**Pull request by andres-blanco**
* Add ConfigurationService in ClientCreateController.php to get
 the correct URL in the form action
 * Fix in AuthCodeRepository to work with Postgres (it doesn't
 accept bools as parameters to querys, they must be converted to
 string)
 * Add missing returns in RoutingService
 * Fix non-compliant jwks endpoint

### CODE REVIEW

File: lib/Services/RoutingService.php [hunk 1]
```diff
@@ -55,13 +55,14 @@ public static function call(string $controllerClassname, bool $authenticated = t
         if ($response instanceof Template) {
             $response->data['messages'] = $container->get(SessionMessagesService::class)->getMessages();
 
-            $response->show();
+            return $response->show();
```
Comment (tvdijen) [inline lib/Services/RoutingService.php]: Template::show() never returns a value, so this is wrong..

File: lib/Services/RoutingService.php [hunk 2]
```diff
@@ -55,13 +55,14 @@ public static function call(string $controllerClassname, bool $authenticated = t
...
+	    
         }
 
         if ($response instanceof ResponseInterface) {
             $emitter = new SapiEmitter();
 
-            $emitter->emit($response);
+            return $emitter->emit($response);
```
Comment (tvdijen) [inline lib/Services/RoutingService.php]: Same

**Comment by andres-blanco** [comment]
I think that with these fixes, everything is working normally using Postgresql as backend.

File: lib/Services/RoutingService.php (hunk 1 above)
Comment (andres-blanco) [reply lib/Services/RoutingService.php]: Fixed in https://github.com/rediris-es/simplesamlphp-module-oidc/pull/36/commits/f799ec1327748d02ee8a4914e19fffdeea4f9b2c

File: lib/Services/RoutingService.php (hunk 2 above)
Comment (andres-blanco) [reply lib/Services/RoutingService.php]: Fixed in https://github.com/rediris-es/simplesamlphp-module-oidc/pull/36/commits/f799ec1327748d02ee8a4914e19fffdeea4f9b2c

File: lib/Services/RoutingService.php (hunk 1 above)
Comment (tvdijen) [reply lib/Services/RoutingService.php]: I don't understand why you want to add the return.. It's of no use..
Anyway, it's up to the maintainer to accept this PR.. I don't have write-access

Comment (andres-blanco) [reply lib/Services/RoutingService.php]: If there is no return then the function throws an exception. 
https://github.com/rediris-es/simplesamlphp-module-oidc/blob/master/lib/Services/RoutingService.php#L67

Comment (tvdijen) [reply lib/Services/RoutingService.php]: Ah, I think I understand.. but it's not this one, but the second case!

Comment (tvdijen) [reply lib/Services/RoutingService.php]: Template::show never returns, so it's safe to remove the return here

Comment (andres-blanco) [reply lib/Services/RoutingService.php]: Without the return after ```show()``` I get this error navigating to ```module.php/oidc/clients/```
```
[Wed Mar 25 17:59:13.778239 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] SimpleSAML\\Error\\Error: UNHANDLEDEXCEPTION, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778481 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Backtrace:, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778564 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 1 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/www/_include.php:17 (SimpleSAML_exception_handler), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778626 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 0 [builtin] (N/A), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778692 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Caused by: SimpleSAML\\Error\\Exception: Response type not supported: SimpleSAML\\XHTML\\Template, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778746 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Backtrace:, referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778798 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 3 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/modules/oidc/lib/Services/RoutingService.php:67 (SimpleSAML\\Modules\\OpenIDConnect\\Services\\RoutingService::call), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778907 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 2 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/modules/oidc/www/clients/index.php:18 (require), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.778967 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 1 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/lib/SimpleSAML/Module.php:254 (SimpleSAML\\Module::process), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.779021 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] 0 /usr/local/app/idp/vendor/simplesamlphp/simplesamlphp/www/module.php:10 (N/A), referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
[Wed Mar 25 17:59:13.779133 2020] [php7:notice] [pid 14] [client 172.24.0.2:36660] simplesamlphp ERR [e069591f73] Error report with id e81c96a1 generated., referer: http://siu:8286/siu/idp/module.php/core/frontpage_federation.php
```
I'm testing in PHP 7.3, maybe it's related?

Comment (tvdijen) [reply lib/Services/RoutingService.php]: I think the issue is (type) confusion..  We're not actually dealing with a `SimpleSAML\XHTML\Template`, but with a `ReflectionClass`...  I believe @sgomez was going to rewrite this part of the module to use Symfony Response-objects instead, so for now I can live with the added returns..

**Comment by restena-sw** [comment]
I have applied the code changes from the PR in its current state and can confirm that the syntax errors are gone and I can use the module's admin interface without errors.

**Comment by sgomez** [comment]
Thanks @andres-blanco. 

Anyway you are trying to solve different things the same PR. I appreciate your comments but as I merged a pending branch that solves some issues you described I can't merge it because there a lot of conflicts. For that reason is a good practice to send separately.

The issue with postgres is something I need to probe because probably fixing this I could create problems with others RDBMS like mysql or sqlite.

I will close this PR and merge RFC 7517 and form route issues separately.

**Comment by andres-blanco** [comment]
Amazing @sgomez ! I know I crammed more than I should in this PR. But I needed it in the product I'm building and it was simpler this way.
Thanks for taking the time to clean up. 

I am really interested in this module, so if you need anything don't hesitate to ask and I will try to help.

**Comment by sgomez** [comment]
@andres-blanco what PHP and postgres version are you using? I can't reproduce the issue. With PHP 7.4 and postgres 10 works great. And I guess that PDO should translate the values.

**Pull request closed without merging**